andela/travel_tool_front
# #161044598 Fix delete checklist item

#### What does this PR do?
Fixes error that allows travel administrators to delete a checklist item without a reason

#### Description of Task to be completed?
- add setState method to clear state on delete checklist item success
- add method to close the delete modal
- update test to accommodate new changes

#### How should this be manually tested?
- Clone the repo - git clone https://github.com/andela/travel_tool_front.git
- Change into the project directory
- Checkout to the branch with git fetch -a then git checkout ft-delete-travel-checklist-item-fix-161044598
- Pull [develop](https://github.com/andela/travel_tool_back) on the backend
- Run the server with make start
- Navigate to Trip Planner page and create multiple requests.
- No request should be deleted without providing a reason regardless of whether any request was deleted previously.

#### Any background context you want to provide?
When a travel administrator deletes a checklist item without a reason for the first time after loading the page, an error is thrown because no reason for deletion was provided.

After the first request is successful, subsequent deletion is successful because the reason for the previous item deleted is still available in the state.

#### What are the relevant pivotal tracker stories?
[#161044598](https://www.pivotaltracker.com/story/show/161044598)

#### Screenshots (if appropriate)
N/A

#### Questions:
N/A
